domain/network/state/linklayer_test.go

```diff
@@ -280,29 +252,33 @@ func (s *linkLayerSuite) TestSetMachineNetConfigNoAddresses(c *tc.C) {
 	// Assert
 	c.Assert(err, tc.ErrorIsNil)
 
-	rows, err := db.QueryContext(ctx, "SELECT name FROM link_layer_device")
+	checkScalarResult(c, db, "SELECT name FROM link_layer_device", "eth0")
+
+	row := db.QueryRowContext(ctx, "SELECT count(*) FROM ip_address")
+	c.Assert(row.Err(), tc.ErrorIsNil)
+
+	var addrCount int
+	err = row.Scan(&addrCount)
+	c.Assert(err, tc.ErrorIsNil)
+	c.Check(addrCount, tc.Equals, 0)
+}
```

Comment:
The `linkLayerBaseSuite` has a method to do this: `checkRowCount`. May or may not be worth updating this suite.